pkg/fanal/types/artifact.go

```diff
@@ -62,19 +65,20 @@ type Layer struct {
 }
 
 type Package struct {
-	ID         string   `json:",omitempty"`
-	Name       string   `json:",omitempty"`
-	Version    string   `json:",omitempty"`
-	Release    string   `json:",omitempty"`
-	Epoch      int      `json:",omitempty"`
-	Arch       string   `json:",omitempty"`
-	Dev        bool     `json:",omitempty"`
-	SrcName    string   `json:",omitempty"`
-	SrcVersion string   `json:",omitempty"`
-	SrcRelease string   `json:",omitempty"`
-	SrcEpoch   int      `json:",omitempty"`
-	Licenses   []string `json:",omitempty"`
-	Maintainer string   `json:",omitempty"`
+	ID         string        `json:",omitempty"`
+	Name       string        `json:",omitempty"`
+	Identifier PkgIdentifier `json:",omitempty"`
```

Comment:
We need to think about `Ref` field.
Perhaps we can mark it as deprecated and upgrade logic to use `PkgIdentifier`.

But it seems that this would be better done in another PR.

Reply:
Yup.. I added a comment at `pkg/types/vulnerability.go` mentioning that's to be deprecated but I took no further action.

Reply:
If I remember correctly, we simply copy `Ref` from `Package` to `DetectedVulnerability`.
It looks more logical to add comment for parent (I mean the first element of this chain (`Package`)).

But I think the best way is to add a comment in both places.

Reply:
done at https://github.com/aquasecurity/trivy/pull/5439/commits/a777f50213bbaafa92cdba195ad419c6b8916274